src/test/BasicTests.hs

```diff
@@ -0,0 +1,139 @@
+-- | This module contains basic tests for 'haskell-source-transformations' 
+module BasicTests
+  ( basicTests
+  )
+where
+
+import           Test.Hspec                     ( Spec
+                                                , Expectation
+                                                , describe
+                                                , it
+                                                , shouldBe
+                                                )
+import           Test.HUnit.Base                ( assertFailure )
+
+import           Control.Monad                  ( void )
+
+import           Language.Haskell.Exts          ( Module
+                                                , ParseResult(..)
+                                                , parseModule
+                                                )
+import           Language.Haskell.Exts.Pretty   ( Pretty
+                                                , prettyPrint
+                                                )
+
+import           Application                    ( processModule
+                                                , specialCons
+                                                )
+import           FreshVars                      ( PMState(..)
+                                                , evalPM
+                                                )
+
+
+basicTests :: Spec
+basicTests = describe "Basic unit tests" tests
+
+-- | Parses a given string to a module and fails if parsing is not
+--   successful.
+parseTestModule :: String -> IO (Module ())
+parseTestModule modStr = case parseModule modStr of
+  ParseOk modul        -> return $ void modul
+  ParseFailed _ errMsg -> assertFailure errMsg
+
+-- | The default state used for pattern matching.
+defaultState :: PMState
+defaultState = PMState { nextId      = 0
+                       , constrMap   = specialCons
+                       , matchedPat  = []
+                       , trivialCC   = False
+                       , opt         = True
+                       , debugOutput = ""
+                       }
+
+-- | Test cases for 'processModule'.
+tests :: Spec
+tests = do
```

Comment:
Since the `BasicTests` module contains only tests for `HST.Application.processModule`, I think the module should be called `HST.ApplicationTests`.